src/soft_assay_rules/local_rule_tester.py

```diff
@@ -81,17 +81,48 @@ def lookup_metadata_json(uuid):
         if Path(fname).exists():
             with open(fname) as infile:
                 json_dict = json.load(infile)
-                LOGGER.debug(f"HERE: {json_dict.keys()}")
             return app_ctx, json_dict
     raise ValueError(f"No cached metadata JSON for {uuid}")
 
 
 def wrapped_lookup_metadata_json(uuid):
-    """Like lookup_entity_json but drop the app_ctx"""
+    """Like lookup_metadata_json but drop the app_ctx"""
     return lookup_metadata_json(uuid)[1]
 
 
-def calculate_assay_info(metadata: dict) -> dict:
+def lookup_rulechain_json(uuid):
+    """
+    Check the directory of cached rulechain json files for the given example, looking for
+    filenames of the form "rulechain_{uuid}_SENNET.json" or ..."_HUBMAP.json".  Pick the
+    first of those found, and infer the uuid's app_ctx from the corresponding string.
+    Return a tuple containing that app_ctx as a string and the JSON dict loaded from
+    the file.  If no such file is found, ValueError is raised.
+    """
+    for app_ctx in ["SENNET", "HUBMAP"]:
+        fname = build_cached_json_fname(uuid, app_ctx,
+                                        dir="captured_rulechain_json",
+                                        prefix="rulechain")
+        if Path(fname).exists():
+            with open(fname) as infile:
+                json_dict = json.load(infile)
+            return app_ctx, json_dict
+    raise ValueError(f"No cached rulechain JSON for {uuid}")
+
+
+def wrapped_lookup_rulechain_json(uuid):
+    """Like lookup_rulechain_json but drop the app_ctx"""
+    return lookup_rulechain_json(uuid)[1]
+
+
+def post_rule_transform(rule_output: dict,
```

Comment:
@jswelling - I thought our intention was to put this logic into a separate rule chain JSON?

Reply:
Yeah, but it's only one line of logic. I can do that easily enough but on actual implementation it seemed kind of silly. How much additional logic do you anticipate? Shall I do it that way?